frontend/src/components/MeetingRoom/UnreadMessagesBadge/UnreadMessagesBadge.spec.tsx

```diff
@@ -0,0 +1,50 @@
+import { describe, it, vi, expect, Mock, beforeEach } from 'vitest';
+import { render, screen } from '@testing-library/react';
+import BiotechIcon from '@mui/icons-material/Biotech';
+import useSessionContext from '../../../hooks/useSessionContext';
+import { SessionContextType } from '../../../Context/SessionProvider/session';
+import UnreadMessagesBadge from './UnreadMessagesBadge';
+import ToolbarButton from '../ToolbarButton';
+
+vi.mock('../../../hooks/useSessionContext');
+const mockUseSessionContext = useSessionContext as Mock<[], SessionContextType>;
+const sessionContext = {
+  unreadCount: 0,
+} as unknown as SessionContextType;
+const LittleButton = () => <ToolbarButton onClick={() => {}} icon={<BiotechIcon />} />;
+
+describe('UnreadMessagesBadge', () => {
+  beforeEach(() => {
+    mockUseSessionContext.mockReturnValue(sessionContext);
+  });
+
+  it('should show unread message number when number is 8', () => {
```

Comment:
Is this shown when there's 8 unread messages?  Or, when there's unread messages period?

If the latter, then maybe spec needs rephrasing, IMO.

Reply:
It's shown when they're there, period. I rephrased it and added another set of matchers to better reflect that.